cg/meta/clean/api.py

```diff
@@ -0,0 +1,111 @@
+import logging
+from datetime import datetime
+from typing import Iterator, List, Optional
+from pathlib import Path
+
+from alchy import Query
+
+from cgmodels.cg.constants import Pipeline
+from housekeeper.store import models as hk_models
+
+from cg.constants.tags import WORKFLOW_PROTECTED_TAGS
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.store import Store, models
+
+LOG = logging.getLogger(__name__)
+
+
+class CleanAPI:
+    def __init__(self, status_db: Store, housekeeper_api: HousekeeperAPI):
+        self.status_db = status_db
+        self.housekeeper_api = housekeeper_api
+
+    def get_bundle_files(
+        self, bundle_name: str, started_at: datetime, pipeline: Pipeline
+    ) -> List[hk_models.File]:
+        """Get any bundle files for a specific version"""
+        hk_bundle_version: Optional[hk_models.Version] = self.housekeeper_api.version(
+            bundle=bundle_name, date=started_at
+        )
+        if not hk_bundle_version:
+            LOG.warning(
+                f"Version not found for "
+                f"bundle:{bundle_name}; "
+                f"pipeline: {pipeline}; "
+                f"date {started_at}"
+            )
+            return []
+
+        LOG.info(
+            f"Version found for "
+            f"bundle:{bundle_name}; "
+            f"pipeline: {pipeline}; "
+            f"date {started_at}"
+        )
+        return self.housekeeper_api.get_files(
+            bundle=bundle_name, version=hk_bundle_version.id
+        ).all()
+
+    @staticmethod
+    def has_protected_tags(file: hk_models.File, protected_tags_lists: List[List[str]]) -> bool:
+        """Check if a file has any protected tags"""
+
+        LOG.info(f"File {file.full_path} has the tags {file.tags}")
+        version_file_tags: List[str] = HousekeeperAPI.get_tag_names_from_file(file)
+
+        _has_protected_tags: bool = False
+        for protected_tags in protected_tags_lists:
+
+            protected_tags_on_file = set(version_file_tags).intersection(protected_tags)
+            if protected_tags_on_file:
+                if protected_tags_on_file == set(protected_tags):
```

Comment:
```suggestion
            if protected_tags_on_file and protected_tags_on_file == set(protected_tags):
```
Less deep - what do you think

Reply:
refactored